prog/weaveproxy/main.go

```diff
@@ -16,13 +16,29 @@ var (
 )
 
 func main() {
-	var target, listen string
-	var withDNS, withIPAM, debug, justVersion bool
+	var (
+		debug       bool
+		justVersion bool
+		listen      string
+		target      string
+		tlsCACert   string
+		tlsCert     string
+		tlsEnabled  bool
+		tlsKey      string
+		tlsVerify   bool
```

Comment:
I suggest reading the `tls*` directly into a `TLSConfig` instance.

It's also worth considering the introduction of a `Config` struct for *all* proxy configuration.